cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedFunction.qll

```diff
@@ -16,6 +16,32 @@ private import TranslatedStmt
  */
 TranslatedFunction getTranslatedFunction(Function func) { result.getAST() = func }
 
+/**
+ * Gets the size, in bytes, of the variable used to represent the `...` parameter in a varargs
+ * function. This is determined by finding the total size of all of the arguments passed to the
+ * `...` in each call in the program, and choosing the maximum of those, with a minimum of 8 bytes.
+ */
+private int getEllipsisVariableByteSize() {
+  result =
+    max(int variableSize |
+      variableSize =
+        max(Call call, int callSize |
+          callSize =
+            sum(int argIndex |
+              call.isEllipsisArgumentIndex(argIndex)
+            |
+              call.getArgument(argIndex).getType().getSize()
+            )
+        |
+          callSize
+        )
+      or
+      variableSize = 8
```

Comment:
Is there a reason the minimum is 8 bytes and not, say, 1 byte?

Reply:
I think I was assuming that we'd be adding alignment padding to any small values passed as varargs, the same way actual code generation would, so I padded the 1 byte out to the size of a pointer. In retrospect, the layout of the buffer probably won't ever be apparent in the IR, so this doesn't matter.